akka-persistence-typed/src/test/java/jdocs/akka/persistence/typed/DurableStatePersistentBehaviorTest.java

```diff
@@ -357,4 +366,91 @@ public CommandHandler<Command, State> commandHandler() {
       // #wrapPersistentBehavior
     }
   }
+
+  interface WithChangeHandler {
+
+    // #changeHandler
+    public class MyPersistentBehavior
+      extends DurableStateBehavior<MyPersistentBehavior.Command, MyPersistentBehavior.State>
+    implements ChangeEventHandler<MyPersistentBehavior.State, MyPersistentBehavior.ChangeEvent>{
```

Comment:
This might be an api design question. I didn't just add the `changeEvent` and `deleteChangeEvent` directly to the DurableStateBehavior for two reasons:
1. nice to have a type for the event, and can't really add a new type parameter to DurableStateBehavior
2. want to see if the handlers are defined or not to be able to fail early if state store is not implementing the needed trait

In scaladsl it's slightly different. No mixin trait because we don't have any base class. It could be the same in javadsl, but maybe a mixin trait is better for Java?

Reply:
I think I change my mind here. For command handler the user has to override `protected def commandHandler()` so it would be a better fit to also have a `withChangeEventHandler`. I'll try.